### PR TITLE
[IMP] web_editor, website: multiple improvements related to new page from template

### PR DESCRIPTION
**Pull request by bso-odoo**
This PR contains multiple improvements related to new page from template:

- generalize default content for blocks so that they also appear during drag'n'drop (instead of patching them specifically so that they work for the "new page" feature)
- simplify code: remove double dialog opening guard, remove async Promise executor
- remove some useless XML in templates

Related: https://github.com/odoo/design-themes/pull/726

task-3555325

### CODE REVIEW

File: addons/website/views/snippets/s_map.xml [hunk 1]
```diff
@@ -2,15 +2,19 @@
...
                     An address must be specified for a map to be embedded
                 </div>
             </div>
-            <!-- GMap iframe will be injected in JS -->
-            <div class="s_map_color_filter"/>
+            <div class="s_map_color_filter">
+                <iframe class="s_map_embedded o_not_editable"
+                    width="100%" height="100%" frameborder="0" scrolling="no" marginheight="0" marginwidth="0"
+                    src="https://maps.google.com/maps?q=250%20Executive%20Park%20Blvd%2C%20Suite%203400%20San%20Francisco%20California%20(US)%20United%20States&amp;t=m&amp;z=12&amp;ie=UTF8&amp;iwloc=&amp;output=embed">
+                </iframe>
```
Comment (bso-odoo) [inline addons/website/views/snippets/s_map.xml:16]: Not a fan of adding this `iframe` in the default content: it gets loaded as soon as we switch to edit mode.
I'd rather go to:
- add a `<t t-set="is_for_drag_and_drop"/>` in `website.snippets`
- add a `t-if="not is_for_drag_and_drop"` on the `iframe` or add a class in that situation that makes it non-rendered, but with a mechanism that removes it when starting to drag the block.

Comment (qsm-odoo) [reply addons/website/views/snippets/s_map.xml:16]: @bso-odoo The task I should start working on again after OXP: https://github.com/odoo/odoo/pull/77460 :wink: There is the data-src on iframe in particular :wink: ... last push last year, I don't remember much. But if you see something small to extract in there... maybe we can merge it with this. If we have the time :grimacing:

**Review by rdeodoo — APPROVED** [review]
@robodoo r+

Thanks for the PR.
As the freeze is tomorrow and this need to be merged, I removed part of the diff, especially the part about `s_preview` as I don't think it's the right way to tackle the issue, since it does not help with `t-snippet-call`, which you actually have to fix in the very specific case of the new page templates.

It's worth a later discussion to find a better way that would make it `t-snippet-call` friendly.
Surely in the direction you suggested about `is_for_drag_and_drop`.

What I [changed](https://github.com/odoo/odoo/compare/cde06f15bdad5ca1bf60a283ce7ab8e3e22cbf59..eed8533229f60fab38e83a29fb9ef4e212ffb9b4):
- Remove the `s_preview` and `s_keep_preview`
- Make the FB snippet t-snippet-call friendly, using `src` instead https://github.com/odoo/odoo/compare/cde06f15bdad5ca1bf60a283ce7ab8e3e22cbf59..eed8533229f60fab38e83a29fb9ef4e212ffb9b4#diff-b2f2f5ef2b41a77e70d68aa12c20bbf09852c0c5327bd83694ef8738624fcc0bR7
- Same for map https://github.com/odoo/odoo/compare/cde06f15bdad5ca1bf60a283ce7ab8e3e22cbf59..eed8533229f60fab38e83a29fb9ef4e212ffb9b4#diff-bc7bc5ccd3241948685be257fa7e453bee69aac6068d45b4265042cca78edeabR12
- For map and FB, it's indeed really not worth losing the `t-snippet-call` capability just to remove the request to those endpoints, as we discussed at the office the other day I think? It's not ideal but the tradeoff is really not worth IMO. It's not going to block any test either (or need to move in `external` testing suite) as we don't rely on those requests.
- Fixed the `ID` added on wrong element for `TOC` snippet https://github.com/odoo/odoo/compare/cde06f15bdad5ca1bf60a283ce7ab8e3e22cbf59..eed8533229f60fab38e83a29fb9ef4e212ffb9b4#diff-5397808e258b9c44c811056afd3deeaf7a44445313593971c2386481f80fcaedR18
- Removed the countdown onbuilt "hack" to sync snippet dropped and preview timers https://github.com/odoo/odoo/compare/cde06f15bdad5ca1bf60a283ce7ab8e3e22cbf59..eed8533229f60fab38e83a29fb9ef4e212ffb9b4#diff-9697bc38c0d3e9c56af409cb640e74154ab0284e5187365e12667975290f2f64L16
It should be done on the SVG preview directly if we want to do that?
- I made countdown t-snippet-call friendly with https://github.com/odoo/odoo/compare/cde06f15bdad5ca1bf60a283ce7ab8e3e22cbf59..eed8533229f60fab38e83a29fb9ef4e212ffb9b4#diff-a0262b81bb090b8c62afbd342b2a30f054cf353a35f9fc0ceeb0f86b7b9cd645R17-R20

What I didn't fully understand and preferred to kept for later:
- The countdown d-block change https://github.com/odoo/odoo/compare/cde06f15bdad5ca1bf60a283ce7ab8e3e22cbf59..eed8533229f60fab38e83a29fb9ef4e212ffb9b4#diff-a0262b81bb090b8c62afbd342b2a30f054cf353a35f9fc0ceeb0f86b7b9cd645R140
- The s_chart empty thing https://github.com/odoo/odoo/compare/cde06f15bdad5ca1bf60a283ce7ab8e3e22cbf59..eed8533229f60fab38e83a29fb9ef4e212ffb9b4#diff-1cc0826abf070db13fa3f173fa6d9209c1c3ee4e1ac164cca8e097c5b460c7d7L121-L128

Could you create a PR once this one is merged for those 2 changes if they are really needed? 

Edit: I had to introduce a new commit for runbot to not fail because of the call to gmap api. This can be changed later in the discussion about the other points.

**Comment by rdeodoo** [comment]
@robodoo rebase-ff

**Comment by rdeodoo** [comment]
@robodoo r+

**Pull request closed without merging**